lib/BuildSystem/BuildSystem.cpp

```diff
@@ -3252,7 +3252,7 @@ class ArchiveShellCommand : public ExternalCommand {
 
   std::vector<std::string> getArgs() const {
     std::vector<std::string> args;
-    args.push_back("ar");
+    args.push_back("llvm-ar");
```

Comment:
I would prefer to conditionalize this on Win32, I'm not sure we should avoid using whatever is named "ar" on other systems.

Reply:
Sure, I can conditionalize it to Win32, but, I do think that we really should prefer to use the LLVM tools.  They give us broader compatibility as the single tool can then be used to cross-compile (e.g. I can build a BSD archive on Linux) which I cannot do otherwise.

Reply:
Cross-compilation is definitely something I'd like to support, but llbuild is general purpose and can easily be used on a system without LLVM/Clang. It'd be a different story if we were directly integrating it to provide the `ar` functionality, but if we are shelling out I'd like to use the standard name.

Reply:
Well, that is the problem - there is no standard name that works across platforms.  Perhaps we should do this the GNU way - use `AR` from the environment and fall back to `ar`?

Reply:
That sounds reasonable to me.